src/test/java/com/puppycrawl/tools/checkstyle/SarifLoggerTest.java

```diff
@@ -94,6 +94,40 @@ public void testAddError() throws IOException {
         verifyContent(getPath("ExpectedSarifLoggerSingleError.sarif"), outStream);
     }
 
+    @Test
+    public void testAddErrorAtColumn1() throws IOException {
+        final SarifLogger logger = new SarifLogger(outStream,
+                OutputStreamOptions.CLOSE);
+        logger.auditStarted(null);
+        final Violation violation =
+                new Violation(1, 1,
+                        "messages.properties", "ruleId", null, SeverityLevel.ERROR, null,
+                        getClass(), "found an error\ntry again");
+        final AuditEvent ev = new AuditEvent(this, "Test.java", violation);
+        logger.fileStarted(ev);
+        logger.addError(ev);
+        logger.fileFinished(ev);
+        logger.auditFinished(null);
+        verifyContent(getPath("ExpectedSarifLoggerSingleErrorColumn1.sarif"), outStream);
```

Comment:
Please extract common part of this 3 method in new method, to show explicitly that that execution is mimic of natural execution in Checker but we do it by low level.

```
public void testAddErrorAtColumn1() throws IOException {
        final SarifLogger logger = new SarifLogger(outStream,
                OutputStreamOptions.CLOSE);
        
        final Violation violation =
                new Violation(1, 1,
                        "messages.properties", "ruleId", null, SeverityLevel.ERROR, null,
                        getClass(), "found an error\ntry again");
        
        executeLogger(this, logger,  "Test.java", violation);
        verifyContent(getPath("ExpectedSarifLoggerSingleErrorColumn1.sarif"), outStream);
    }
```